src/coreclr/nativeaot/Runtime/gcheaputilities.cpp

```diff
@@ -47,6 +47,20 @@ extern "C" HRESULT GC_Initialize(
 
 #ifndef DACCESS_COMPILE
 
+CrstStatic g_eventStashLock;
```

Comment:
Can this be in clrgc.enabled.cpp to make it more pay-for-play?

Reply:
I moved `GCHeapUtilities::RecordEventStateChange` to be defined in both `clrgc.disabled.cpp` and `clrgc.enabled.cpp`. When GC loading is disabled, there is no difference from the status quo.